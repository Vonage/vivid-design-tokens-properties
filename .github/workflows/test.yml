name: Test Design Token Properties Flow

on:
  push:
    tags:
      - test*
  
jobs:
  publish:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2
        with:
          ref: master

      - name: Setup NodeJS 14
        uses: actions/setup-node@v1
        with:
          node-version: 14
          registry-url: 'https://npm.pkg.github.com'

      - name: Fetch data
        run: npm run fetch
        env:
		  FIGMA_FILE_TOKEN: ${{ secrets.FIGMA_FILE_TOKEN }}
          FIGMA_AUTH_TOKEN: ${{ secrets.FIGMA_AUTH_TOKEN }}