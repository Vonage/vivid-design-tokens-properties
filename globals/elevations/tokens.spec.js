import elevationTokens from './tokens';
import StyleDictionaryPackage from 'style-dictionary';
import {getStyleDictionaryConfig} from "../../test/test-dictionary-coherent";

describe(`Elevation Tokens`, function () {
    it(`should create the tokens`, function () {
        expect(JSON.stringify(elevationTokens, null, 2)).toMatchSnapshot();
    });

    it(`should transpile in styledDictionary`, function () {
        for (const scheme of ['dark', 'light']) {
            for (const alt of ['alternate', 'main']) {
                const config = getStyleDictionaryConfig([
                    `./globals/elevations/tokens.js`,
                    `./elevations/${scheme}/${alt}.json`
                ], 'elevations.scss');
                expect(() => StyleDictionaryPackage.extend(config).buildPlatform('web')).not.toThrow();
            }
        }
    });
});
