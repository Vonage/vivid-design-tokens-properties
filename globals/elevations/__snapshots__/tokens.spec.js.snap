// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elevation Tokens should create the tokens 1`] = `
"{
  \\"test\\": {
    \\"elevation\\": {
      \\"dp-2\\": {
        \\"canvas\\": {
          \\"value\\": \\"{scheme.elevation.dp-2.canvas.value}\\"
        },
        \\"filter\\": {
          \\"value\\": \\"{scheme.elevation.dp-2.filter.value}\\"
        }
      },
      \\"dp-4\\": {
        \\"canvas\\": {
          \\"value\\": \\"{scheme.elevation.dp-4.canvas.value}\\"
        },
        \\"filter\\": {
          \\"value\\": \\"{scheme.elevation.dp-4.filter.value}\\"
        }
      },
      \\"dp-8\\": {
        \\"canvas\\": {
          \\"value\\": \\"{scheme.elevation.dp-8.canvas.value}\\"
        },
        \\"filter\\": {
          \\"value\\": \\"{scheme.elevation.dp-8.filter.value}\\"
        }
      },
      \\"dp-12\\": {
        \\"canvas\\": {
          \\"value\\": \\"{scheme.elevation.dp-12.canvas.value}\\"
        },
        \\"filter\\": {
          \\"value\\": \\"{scheme.elevation.dp-12.filter.value}\\"
        }
      },
      \\"dp-16\\": {
        \\"canvas\\": {
          \\"value\\": \\"{scheme.elevation.dp-16.canvas.value}\\"
        },
        \\"filter\\": {
          \\"value\\": \\"{scheme.elevation.dp-16.filter.value}\\"
        }
      },
      \\"dp-24\\": {
        \\"canvas\\": {
          \\"value\\": \\"{scheme.elevation.dp-24.canvas.value}\\"
        },
        \\"filter\\": {
          \\"value\\": \\"{scheme.elevation.dp-24.filter.value}\\"
        }
      }
    }
  }
}"
`;
